### PR TITLE
[semver:minor] Go build

### PR DESCRIPTION
**Pull request by corrjo**
Adds a gox-build and test job

### CODE REVIEW

**Comment by ahammond** [comment]
build-lambdas?

File: src/commands/gox-build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+parameters:
+  source_dir:
+    type: string
+    default: "lambdas"
+    description: "directory which contains your go code"
+steps:
+  - run:
+      name: go get gox
+      command: |
+        which gox || go get -u github.com/mitchellh/gox
```
Comment (ahammond) [inline src/commands/gox-build.yml:10]: 🦖 🌮

File: src/commands/test.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    type: enum
+    enum: ['acceptance', 'unit']
+steps:
+  - aws-cli/setup
+  - run:
+      name: install ginkgo
+      command: |
+        if [ -z "$(which ginkgo)" ]; then
+          echo "Installing ginkgo"
+          go get github.com/onsi/ginkgo/ginkgo
```
Comment (ahammond) [inline src/commands/test.yml:15]: If this is intended to support `hss` based legacy systems, we need to fetch the legacy version of ginkgo. If it is not, then it must be clearly marked as non-compatible with legacy systems.

Also, we have been super-lax with the READMEs / docs in all our orbs so far... and I guess this is the inflection point which drives us to finally do something on that front.

File: src/jobs/test.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+description: >
+  Run tests using the myhelix test script for go projects
```
Comment (ahammond) [inline src/jobs/test.yml:2]: I don't love the pattern of having a `tests.sh` file in the repos. But... I also kinda love the idea of making it easier to adopt circleci and move forward.

**Review by ahammond — APPROVED** [review]

File: src/jobs/test.yml (hunk 3 above)
Comment (corrjo) [reply src/jobs/test.yml:2]: Ah, I meant to change that, thats from another branch where I'm using an embedded test.sh in the orb to run the tests. this version of the orb just calls ginkgo directly.

**Pull request merged**